lib/iris/fileformats/name_loaders.py

```diff
@@ -78,11 +78,11 @@ def read_header(file_handle):
     header = {}
     header['NAME Version'] = next(file_handle).strip()
     for line in file_handle:
-        words = line.split(':', 1)
+        words = line.split(b':', 1)
         if len(words) != 2:
             break
         key, value = [word.strip() for word in words]
-        header[key] = value
+        header[key.decode()] = value
```

Comment:
Since the keys are always strings, I'm not sure it makes much difference in this case. Python 2 is _very_ lax, e.g., `'foo' in {u'foo': 1} == True`.